source/common/http/filter/fault_filter.cc

```diff
@@ -142,6 +150,21 @@ void FaultFilter::abortWithHTTPStatus() {
   callbacks_->requestInfo().setResponseFlag(Http::AccessLog::ResponseFlag::FaultInjected);
 }
 
+bool FaultFilter::matchesTargetCluster(HeaderMap& headers) {
+  bool matches = true;
+
+  if (!config_->targetCluster().empty()) {
+    // TODO PERF Eliminate repeated route matching
```

Comment:
not sure if this is worth optimizing, router filter is the main consumer here, there are not many other filters who wish to recalculate routes.

defer this to @mattklein123